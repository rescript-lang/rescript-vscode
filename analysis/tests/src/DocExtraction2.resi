/*** Module level doc here.*/

/** Type t is pretty cool.*/
type t

/** Makerz of stuffz. */
let make: unit => t

module InnerModule: {
  /*** This inner module is nice...*/

  /** This type is also t. */
  type t

  /** Maker of tea.*/
  let make: unit => t
}

// ^dex
