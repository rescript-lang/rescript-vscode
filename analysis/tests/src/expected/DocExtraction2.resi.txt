Documentation extraction src/DocExtraction2.resi
extracting docs for src/DocExtraction2.resi

{
  "name": "DocExtraction2",
  "docstrings": ["Module level doc here."],
  "items": [
  {
    "id": "DocExtraction2.t",
    "kind": "type",
    "name": "t",
    "signature": "type t",
    "docstrings": ["Type t is pretty cool."],
    "linkables": []
  }, 
  {
    "id": "DocExtraction2.make",
    "kind": "value",
    "name": "make",
    "signature": "let make: unit => t",
    "docstrings": ["Makerz of stuffz."],
    "linkables": [{
        "linkId": "DocExtraction2.t",
        "name": "t",
        "external": false
      }]
  }, 
  {
    "id": "InnerModule.DocExtraction2",
    "kind": "module",
    "item": 
    {
      "name": "InnerModule",
      "docstrings": [],
      "items": [
      {
        "id": "DocExtraction2.InnerModule.t",
        "kind": "type",
        "name": "t",
        "signature": "type t",
        "docstrings": ["This type is also t."],
        "linkables": []
      }, 
      {
        "id": "DocExtraction2.InnerModule.make",
        "kind": "value",
        "name": "make",
        "signature": "let make: unit => t",
        "docstrings": ["Maker of tea."],
        "linkables": [{
            "linkId": "DocExtraction2.InnerModule.t",
            "name": "t",
            "external": false
          }]
      }]
    }
  }]
}

